src/ir/global-utils.h

```diff
@@ -64,7 +64,7 @@ inline bool canInitializeGlobal(const Expression* curr) {
   }
   return Properties::isSingleConstantExpression(curr) ||
          curr->is<GlobalGet>() || curr->is<RttCanon>() || curr->is<RttSub>() ||
-         curr->is<StructNew>();
+         curr->is<StructNew>() || curr->is<ArrayNew>() || curr->is<I31New>();
```

Comment:
Shouldn't we be walking or recursing into the operands to e.g. struct.new to make sure none of the operands are disallowed in global initializers? Previously we only allowed constant i31.new, but now we would allow an i31.new wtth a multiplication or similar as its operand.

Reply:
Good point, this was inconsistent - we checked I31 children but not Struct ones. Fixed to check all children.

(Removed a `const*` to make this compile - keeping the const would require refactoring the Iterator class. Not sure that's worth it, given the const here doesn't give much benefit, not sure why it was there.)